### PR TITLE
Update quickstart documentation

### PR DESCRIPTION
**Pull request by RobbeSneyders**
Contributes towards #1531

Updates the quickstart documentation for v3. I removed some more detailed stuff that I don't think belongs on this page.

### CODE REVIEW

**Review by Ruwann — APPROVED** [review]
Looks very nice in the preview! Some minor additions and questions, but looks very nice in general with the different tabs

File: docs/index.rst [hunk 1]
```diff
@@ -57,6 +57,7 @@ Documentation
    :maxdepth: 2
 
    quickstart
+   apps_and_apis
```
Comment (Ruwann) [inline docs/index.rst:60]: remnant from another PR perhaps?

File: docs/conf.py [hunk 2]
```diff
@@ -27,21 +27,14 @@
...
-                   'show-module-summary',
-                   'special-members',
-                   'imported-members']
-autoapi_python_class_content = 'both'
-autoapi_dirs = [
-    '../connexion'
+extensions = [
+    'sphinx.ext.autodoc',
+    'sphinx_copybutton',
+    'sphinx_design',
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply docs/conf.py:33]: `sphinx_design` is the extension that offers the functionality for the tabs and footers. I used two separate packages for this at first, but I like the look of `sphinx_design` better.

Just not completely happy with the color palette yet, but that's something we can customize (in a separate PR).

File: connexion/middleware/main.py [hunk 3]
```diff
@@ -358,6 +358,13 @@ def add_api(
     def add_error_handler(
         self, code_or_exception: t.Union[int, t.Type[Exception]], function: t.Callable
     ) -> None:
+        """
+        Register a callable to handle application errors.
+
+        :param code_or_exception: An exception class or the status code of HTTP exceptions to
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply connexion/middleware/main.py:364]: I actually think plural works best here

File: docs/index.rst (hunk 1 above)
Comment (RobbeSneyders) [reply docs/index.rst:60]: Good catch

File: pyproject.toml [hunk 4]
```diff
@@ -78,6 +78,8 @@ pytest-cov = "~2.12.1"
 [tool.poetry.group.docs.dependencies]
 sphinx = "5.3.0"
 sphinx-autoapi = "2.0.1"
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply pyproject.toml:80]: Maybe not, I'll try :)

File: docs/quickstart.rst [hunk 5]
```diff
@@ -1,116 +1,280 @@
...
+                :noindex:
+
+The Swagger UI
 --------------
-By default connexion uses the default flask server but you can also use Tornado_ or gevent_ as the HTTP server, to do so set server
-to ``tornado`` or ``gevent``:
 
-.. code-block:: python
+If you installed connexion using the :code:`swagger-ui` extra, a Swagger UI is available for each
+API, providing interactive documentation. By default the UI is hosted at :code:`{base_path}/ui/`
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply docs/quickstart.rst:236]: Yes, I think we can have a separate page on the Swagger UI and link to it here. But we'll need to add it first.

**Pull request merged**